src/core/user/proc.rs

```diff
@@ -669,16 +673,30 @@ mod tests {
 
     #[test]
     fn get_threads() -> Result<()> {
-        let (tx, rx) = mpsc::channel();
+        let cnt = Arc::new(Mutex::new(0));
```

Comment:
I would suggest using std::sync::Barrier (waiting from both).
Optional nit: using join() would be nice as well

Reply:
no, I don't see any. Feel free to ignore the optional one.